### ISSUE TITLE
Use of iOS 13 shortcut app

### ISSUE THREAD

**Issue by Trebastian** [TelegramMessenger/Telegram-iOS#190]
Would it be possible to implement Features like "Send Message" to iOS Shortcut App like the IMessage app has? It would be great in combination with e.g. a telegram bot.

**Comment by kharmadar** [comment]
Even implementing basic x-callback URLs will be hell of a help

**Comment by NightMachinary** [comment]
Relevant issue: https://github.com/peter-iakovlev/Telegram/issues/309
If the URIs are fixed, the shortcuts can be created without further support, though the UX won’t be good.

**Comment by jazzcrack** [comment]
Any news on this topic? 🤔

**Comment by iMonZ** [comment]
+1

I would like to send files via shortcut!

I want to create a shortcut where I could send pictures as a file without exporting them to my ICloud Drive and then selecting it inside of telegram (This is the only way to send files without quality loss in iOS!)

**Comment by kastnerorz** [comment]
Any news? 👀

**Comment by carbongo** [comment]
Any chance of getting that feature? Please?

**Comment by carbongo** [comment]
There are actually some awesome potential use cases:
- Text about your arrival/leaving home (useful for a family)
- Set up bot-less time-based messages
- Send a photo you just took
- Forward e-mails
and much more

**Comment by maswellM5** [comment]
I’d love to see this as well please!  So many useful applications for it.

**Comment by Budhill** [comment]
Yes it would be so cool to have!

**Comment by 1onar** [comment]
Shortcuts please! 🙏🏼

**Comment by vstepaniuk** [comment]
When I add a single action from suggested ones to call a certain contact, and execute it, it shows an error.

**Comment by forenheith** [comment]
Btw, iOS 14 doesn’t work as well. I think currently we have the only way to solve this - use a Pythonista  script

**Comment by MorcosS** [comment]
A way to send live location via telegram using shortcuts can be a great add to this.

**Comment by mxmaxime** [comment]
Hi! Same here, it would be very useful.

**Comment by oluckyman** [comment]
“Send a message” shortcut opens endless possibilities with telegram bots. 
I have a bot which I use daily. now it takes six steps:
1. find telegram app
2. open telegram app
3. find the chat with the bot 
4. open chat with the bot
5. type a message (sometimes have to choose proper language in the keyboard)
6. send a message

with “send a message” shortcut it could be one voice command. 

I have concern about Durov’s opinion about Apple. Maybe it affects the priorities between iOS and android versions, or they just use androids more and see no value in shortcuts.

**Comment by oluckyman** [comment]
found a workaround!
**Open URLs** shortcut that opens URL
tg://msg?text=/today&to=@yogawithadrienebot

it reduces 6 steps to 2

Format of the URL got from #445 issue:
`tg://msg?text=MESSAGE&to=@USERNAME_OR_PHONE`

**Comment by francescosche** [comment]
So.. No news about this?
I see you can now send messages to your private contacts, but not bots or groups :(

**Comment by kaagh** [comment]
Sending messages to bots would be very useful. Write whenever there will be updates about this. That's the main reason why this issue was opened.

**Comment by j4hangir** [comment]
> found a workaround!
> 
> **Open URLs** shortcut that opens URL
> 
> [tg://msg?text=/today&to=@yogawithadrienebot](tg://msg?text=/today&to=@yogawithadrienebot)
> 
> it reduces 6 steps to 2
> 
> 
> 
> Format of the URL got from #445 issue:
> 
> `tg://msg?text=MESSAGE&to=@USERNAME_OR_PHONE`

This looks promising, but doesn't actually do anything for me, just opens tg.

**Comment by lukiiimohhU** [comment]
non-updated since [2021](https://github.com/TelegramMessenger/Telegram-iOS/issues/190#issuecomment-974632180)

**Comment by lance36** [comment]
+1
still unable to send messages to bot from shortcuts. PLEASE <3

**Comment by SidShaytay** [comment]
@laktyushin - Any plans around this? I'm wiring in an agentic automation within iOS and not being able to message a Telegram bot (=team coordinator) is a hurdle to get Telegram into the workflows. In addition, you guys should ship some built-in Siri shortcuts to streamline this even more.
